Dockerfile

```diff
@@ -1,24 +1,21 @@
-FROM ubuntu:16.04
-
-ARG DEBIAN_FRONTEND=noninteractive
-ARG PIP_NO_CACHE_DIR=0
-
-RUN \
-    # Update the Ubuntu package index
-       apt-get update \
-    # Install the required Ubuntu packages
-    && apt-get install -y \
-        fontconfig \
-        python-pip \
-    # Remove the APT cache to reduce Docker image size
-    && rm -rf /var/lib/apt/lists/* \
-    # Install Pipenv
-    && pip install pipenv
+# FROM ubuntu:16.04
+FROM python:2-slim
```

Comment:
There's actually 2 reasons I've considered going with `ubuntu:16.04` Docker image instead of `python:2.7` and its derivatives/alternatives:

1. `ubuntu:18.04` and newer versions of Debian (which `python:2.7` and most of its derivatives is based upon) come with newer/customized versions of GNU C headers that break the `pycparser` Python package, and `smoke_test` in return

1. during my tests I've found `ubuntu:16.04` based Docker image to be smaller than that of the `python:2.7` based Docker image

Please verify that:

- [ ] `python:2.7-slim` based Docker image is smaller than that of the `ubuntu:18.04` based Docker image. If possible please supply the image sizes, for archival purposes
- [ ] smoke_test runs correctly for the dockerized T12 template: `$ ./dockerized.sh init template=T12 assignment-pack init-revert`
- [ ] smoke_test runs correctly for the dockerized T34 template: `$ ./dockerized.sh init template=T34 assignment-pack init-revert`
- [ ] smoke_test runs correctly for the dockerized SOV template: `$ ./dockerized.sh init template=SOV assignment-pack init-revert`

Reply:
My bad - when testing image sizes please use `ubuntu:16.04` for comparison, instead of `ubuntu:18.04`.

Reply:
Here are the image size differences. No that significant, though further size optimisation may be possible.

REPOSITORY | TAG    | SIZE
---------- | ------ | -----
pjisp      | 16.04  | 565MB
pjisp      | 2-slim | 523MB

As for the tests, I have ran all three on both images, but with the added `test-solution` target as well. All tests passed. So far no smoke.